test/Interop/Cxx/destructor/basic.swift

```diff
@@ -0,0 +1,61 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXX Destructor")
+
+@inline(never)
+func createTypeWithUserProvidedDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+}
+
+@inline(never)
+func createTypeWithEmptyDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let empty = EmptyDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialImplicitDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let implicit = HasNonTrivialImplicitDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialDefaultDestructor(with ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let def = HasNonTrivialDefaultDestructor(box: box)
+}
+
+CXXDestructorTestSuite.test("Basic object with destructor") {
+  let ptr = UnsafeMutablePointer<Int32>.allocate(capacity: 1)
+  ptr.pointee = 0
+  createTypeWithUserProvidedDestructor(with: ptr)
+  expectEqual(ptr.pointee, 42)
+}
+
+CXXDestructorTestSuite.test("Nested objects with destructors") {
+  let ptr = UnsafeMutablePointer<Int32>.allocate(capacity: 1)
+  ptr.pointee = 0
+  createTypeWithEmptyDestructor(with: ptr)
+  expectEqual(ptr.pointee, 42)
+}
+
+CXXDestructorTestSuite.test("Implicit destructor, member with user-defined destructor") {
+  let ptr = UnsafeMutablePointer<Int32>.allocate(capacity: 1)
+  ptr.pointee = 0
+  createTypeWithNonTrivialImplicitDestructor(with: ptr)
+  expectEqual(ptr.pointee, 42)
+}
+
+CXXDestructorTestSuite.test("Default destructor, member with user-defined destructor") {
+  let ptr = UnsafeMutablePointer<Int32>.allocate(capacity: 1)
+  ptr.pointee = 0
+  createTypeWithNonTrivialDefaultDestructor(with: ptr)
+  expectEqual(ptr.pointee, 42)
+}
```

Comment:
Could you also add some tests that exercise the witness table entry, for example, tests with existentials that contain these C++ types, and tests that pass these C++ types as generic type parameters.